core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -81,11 +82,18 @@ abstract class AppBase(
     val stageInfoDetails = stageInfo.info.details
     val mlOps = if (stageInfoDetails.contains(MlOps.sparkml) ||
       stageInfoDetails.contains(MlOps.xgBoost)) {
-      // Check if it's a pyspark eventlog
-      mlEventLogType = if (stageInfoDetails.contains(MlOps.pysparkLog)) {
-        MlOpsEventLogType.pyspark
+
+      // Check if it's a pyspark eventlog and set the mleventlogtype to pyspark
+      // Once it is set, do not change it to scala if other stageInfoDetails don't match.
+      if (stageInfoDetails.contains(MlOps.pysparkLog)) {
+        if (!pysparkLogFlag) {
+          mlEventLogType = MlOpsEventLogType.pyspark
+          pysparkLogFlag = true
+        }
       } else {
-        MlOpsEventLogType.scala
+        if (!pysparkLogFlag) {
+          mlEventLogType = MlOpsEventLogType.scala
+        }
```

Comment:
The default type for`mlEventLogType` is empty string.
If none of the stages capture the eventLogType, then the Qualification would not define it. Is that ok?
The reason I am asking is that when I was comparing different spark versions, I found that some execs could be moved out of WholeStageCodeGen (i.e., broadcastExchange could cause this)
Then this means that they won't belong to a specific stage.

Reply:
> If none of the stages capture the eventLogType, then the Qualification would not define it. Is that ok?

The code is such that mlEventLogType is set to `scala` if there is no  `py4j.GatewayConnection.run` in any of the stages in the application. So it won't be empty string once this function is called. If there is atleast one stage which has this string (py4j.GatewayConnection.run) then we classify that eventlog as pyspark for that app and don't override   `mlEventLogType ` after that. 
For example, In stage 1 there is no `py4j.GatewayConnection.run` so mlEventLogType is set to `scala`. In the next stage  lets assume `py4j.GatewayConnection.run` is found so `mlEventLogType` is set to `pyspark` and never changed.
Please let me know if this makes sense or if you see any issues with this approach.